### PR TITLE
Email subscription component

### PR DESCRIPTION
**Pull request by w-gao**
Problem
=======
Part of [C2D-27](https://cruzhacks-dev.atlassian.net/browse/C2D-27). Modified from last year's email subscription component. 

Solution
========
What I/we did to solve this problem
* Add EmailSubscription component

Change Summary:
---------------
* Add basic email subscription component from last year

Dev-Ops
=======
If you added an ENV variable, please check off that you've updated the following  
- [ ] Key & Sample value to `.env` & `sample.env` 
- [ ] GCP Secret Manager (Both development and production)
- [ ] Github Secrets (Added a development and production variable)
- [ ] Github Workflows `.github/workflows/dev.yml` & `.github/workflows/prod.yml`
- [ ] webpack-config.js

Images/Important Notes (optional):
-----------------------
* Insert any notes/issues, dependencies added or removed, or images

### CODE REVIEW

**Review by Alperdec — APPROVED** [review]
LGTM

File: src/components/EmailSubscription/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const SubmissionStates = {
+  NotSubmitted: 0,
+  Loading: 1,
+  Submitted: 2,
+  Errored: 3,
+}
+
+interface EmailSubscriptionProps {}
+
+export const EmailSubscription: React.FC<EmailSubscriptionProps> = () => {
```
Comment (krishnakpandian) [inline src/components/EmailSubscription/index.tsx]: Should we keep this empty interface or just leave it as `React.FC`?

**Review by krishnakpandian — APPROVED** [review]
Looks Good!

File: src/components/EmailSubscription/index.tsx (hunk 1 above)
Comment (w-gao) [reply src/components/EmailSubscription/index.tsx]: Oh yeah, I was planning to define some props, but I don't think we need this anymore.

File: src/components/EmailSubscription/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+export const EmailSubscription: React.FC = () => {
+  const [email, setEmail] = useState("")
+  const [state, setState] = useState(SubmissionStates.NotSubmitted)
+  const [message, setMessage] = useState("")
+
+  const submitEmail = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setState(SubmissionStates.Loading)
+    setMessage("Please wait while your message is being submitted...")
```
Comment (krishnakpandian) [inline src/components/EmailSubscription/index.tsx]: One thing, I just remembered really late but can we validate that the form data is not an empty string and is valid email [regex](https://www.w3resource.com/javascript/form/email-validation.php).

**Review by krishnakpandian — CHANGES_REQUESTED** [review]

File: src/components/EmailSubscription/index.tsx (hunk 2 above)
Comment (w-gao) [reply src/components/EmailSubscription/index.tsx]: Will do!

Comment (w-gao) [reply src/components/EmailSubscription/index.tsx]: This has been added, but will probably be best to test with #45.

**Review by krishnakpandian — APPROVED** [review]

**Pull request merged**